crates/oxc_semantic/src/builder.rs

```diff
@@ -35,6 +35,17 @@ struct UnusedLabels<'a> {
     labels: Vec<AstNodeId>,
 }
 
+#[derive(Debug, Clone)]
+pub struct VariableInfo {
+    pub name: Atom,
+    pub span: Span,
+}
+
+#[derive(Debug)]
+pub struct RedeclareVariables {
+    pub variables: FxHashMap<Atom, VariableInfo>,
```

Comment:
Using a hashmap doesn't look correct, happens when you have multiple redeclarations in difference scopes?

Reply:
If you save the redeclared atom/span, and also its symbol_id, then you should be able to loop over this and print things accordingly.

Reply:
But the case that same declarations in different scope and the first declaration couldn't be found (like `function f() { var a; { var a; } } `) or the global variable case too.